File: node/src/consumable/crm.ts
import nodemailer from "nodemailer";

const { EMAIL_USER, EMAIL_PASS } = process.env;

const transporter = nodemailer.createTransport({
  service: "Outlook365",
  auth: {
    user: EMAIL_USER,
    pass: EMAIL_PASS,
  },
});

const enviarEmail = (email: string) => {
  transporter.sendMail(
    {
      from: EMAIL_USER,
      to: email,
      subject: "MANUALL: Você recebeu uma mensagem!",
      text: "Nosso assistente virtual Manuel te enviou uma mensagem",
    },
    (err) => {
      if (err) {
        console.log("Erro no envio de email para o usuário ", email);
        console.log(err);
      }
    }
  );
};

export default enviarEmail;

File: go/consumable/crm.go
package consumable

import (
	"fmt"
	"net/smtp"
	"os"
)

func Crm(to string) {
	from := os.Getenv("EMAIL_ADDRESS")
	password := os.Getenv("EMAIL_PASS")

	smtpServer := "smtp.gmail.com"
	smtpPort := 587

	subject := "MANUALL: Voce recebeu uma mensagem!"
	body := "Nosso assistente virtual Manuel te enviou uma mensagem"

	message := fmt.Sprintf("Subject: %s\r\n\r\n%s", subject, body)

	auth := smtp.PlainAuth("", from, password, smtpServer)

	err := smtp.SendMail(fmt.Sprintf("%s:%d", smtpServer, smtpPort), auth, from, []string{to}, []byte(message))
	if err != nil {
		fmt.Println("Error sending email:", err)
		return
	}
}
